accepted/2020/windows-specific-apis/windows-specific-apis.md

```diff
@@ -0,0 +1,475 @@
+
+* **HttpListenerTimeoutManager**
+    - `set_EntityBody(TimeSpan)`
+    - `set_HeaderWait(TimeSpan)`
+    - `set_MinSendBytesPerSecond(Int64)`
+    - `set_RequestQueue(TimeSpan)`
+* **IPEndPoint**
+    - `Create(SocketAddress)`
+* **SocketAddress**
+    - `.ctor(AddressFamily)`
+    - `.ctor(AddressFamily, Int32)`
+    - `get_Family()`
+    - `ToString()`
+
+### System.Net.Sockets
+
+* **Socket**
+    - `AcceptAsync(SocketAsyncEventArgs)`
+    - `BeginAccept(Int32, AsyncCallback, Object)`
+    - `BeginAccept(Socket, Int32, AsyncCallback, Object)`
+    - `BeginReceiveFrom(Byte[], Int32, Int32, SocketFlags, EndPoint, AsyncCallback, Object)`
+    - `BeginReceiveMessageFrom(Byte[], Int32, Int32, SocketFlags, EndPoint, AsyncCallback, Object)`
+    - `IOControl(Int32, Byte[], Byte[])`
+    - `IOControl(IOControlCode, Byte[], Byte[])`
+    - `ReceiveFrom(Byte[], EndPoint)`
+    - `ReceiveFrom(Byte[], Int32, Int32, SocketFlags, EndPoint)`
+    - `ReceiveFrom(Byte[], Int32, SocketFlags, EndPoint)`
+    - `ReceiveFrom(Byte[], SocketFlags, EndPoint)`
+    - `ReceiveFromAsync(SocketAsyncEventArgs)`
+    - `ReceiveMessageFrom(Byte[], Int32, Int32, SocketFlags, EndPoint, IPPacketInformation)`
+    - `ReceiveMessageFromAsync(SocketAsyncEventArgs)`
+    - `SetIPProtectionLevel(IPProtectionLevel)`
+* **SocketTaskExtensions**
+    - `AcceptAsync(Socket)`
+    - `AcceptAsync(Socket, Socket)`
+    - `ReceiveFromAsync(Socket, ArraySegment<Byte>, SocketFlags, EndPoint)`
+    - `ReceiveMessageFromAsync(Socket, ArraySegment<Byte>, SocketFlags, EndPoint)`
+* **TcpListener**
+    - `AllowNatTraversal(Boolean)`
+* **TransmitFileOptions**
+    - `Disconnect`
+    - `ReuseSocket`
+    - `UseKernelApc`
+    - `UseSystemThread`
+    - `WriteBehind`
+* **UdpClient**
+    - `AllowNatTraversal(Boolean)`
+
+### System.Runtime.InteropServices
+
+* **DispatchWrapper**
+    - `get_WrappedObject()`
+* **Marshal**
+    - `AddRef(IntPtr)`
+    - `BindToMoniker(String)`
+    - `ChangeWrapperHandleStrength(Object, Boolean)`
+    - `CreateAggregatedObject(IntPtr, Object)`
+    - `CreateAggregatedObject<T>(IntPtr, T)`
+    - `FinalReleaseComObject(Object)`
+    - `GetComInterfaceForObject(Object, Type)`
+    - `GetComInterfaceForObject(Object, Type, CustomQueryInterfaceMode)`
+    - `GetComInterfaceForObject<T, TInterface>(T)`
+    - `GetIUnknownForObject(Object)`
+    - `GetNativeVariantForObject(Object, IntPtr)`
+    - `GetNativeVariantForObject<T>(T, IntPtr)`
+    - `GetObjectForIUnknown(IntPtr)`
+    - `GetObjectForNativeVariant(IntPtr)`
+    - `GetObjectForNativeVariant<T>(IntPtr)`
+    - `GetObjectsForNativeVariants(IntPtr, Int32)`
+    - `GetObjectsForNativeVariants<T>(IntPtr, Int32)`
+    - `GetStartComSlot(Type)`
+    - `GetTypedObjectForIUnknown(IntPtr, Type)`
+    - `GetTypeFromCLSID(Guid)`
+    - `GetTypeInfoName(ITypeInfo)`
+    - `GetUniqueObjectForIUnknown(IntPtr)`
+    - `QueryInterface(IntPtr, Guid, IntPtr)`
+    - `Release(IntPtr)`
+    - `ReleaseComObject(Object)`
```

Comment:
there are few more methods in `Marshal` that throw `PlatformNotSupportedException`:

* `CreateWrapperOfType(object, Type)`
* `CreateWrapperOfType<T, TWrapper>(T)`
* `GetComObjectData(object,object)`
* `SetComObjectData(object,object,object)`
* `GetIDispatchForObject(object)`

https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.NoCom.cs

Reply:
Added